### PR TITLE
Expose the existing COMMENT_ENTERED_MODERATION_QUEUE event as a webhook

### PR DESCRIPTION
**Pull request by ddrysdale-ffx**
Exposing this event as a webhook will allow clients to be notified when a comment is created in PREMOD status in the PENDING/UNMODERATED moderation queues.

PR corresponding to issue - https://github.com/coralproject/talk/issues/4668 

## These changes will impact:

- [ ] commenters
- [ ] moderators
- [X] admins
- [X] developers

## What changes to the GraphQL/Database Schema does this PR introduce?

- Adds COMMENT_ENTERED_MODERATION_QUEUE to the WEBHOOK_EVENT_ENUM enum

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indicies.

## How do I test this PR?

- copied from similar PR for COMMENT_LEFT_MODERATION_QUEUE - https://github.com/coralproject/talk/pull/4380
- spin up a quick express + cors middleware web server to act as a web hook listener, the following script works nicely:

```ts
const express = require("express");
const cors = require("cors");

const PORT = 7000;
const HOST = "localhost";

const run = async () => {
  const app = express();

  app.use(cors());
  app.use(express.json());

  app.post("/listen", (req, res) => {
    const { headers, body } = req;

    console.log(headers, body);

    res.sendStatus(200);
  });

  app.get("/", (req, res) => {
    res.status(200).send("coral-webhook-listener");
  });

  app.listen(PORT, HOST, () => {
    console.log(`coral-webhook-listener is listening on ${HOST}:${PORT}...`);
  });
};

void run();
```

- start up Coral
- go to Admin > Configure > Webhooks and click + ADD WEBHOOK ENDPOINT
- for Endpoint URL use: http://localhost:7000/listen
- check on `COMMENT_ENTERED_MODERATION_QUEUE`
- click ADD WEBHOOK ENDPOINT
- create some comments (as a commenter - ie not staff)
- observe webhook events on the listener and with a similar payload to below but status NONE for both PENDING and UNMODERATED queues
- go to Admin > Configure > Moderation and turn On "Pre-moderate all comments"
- create some more comments
- observe webhook events on the listener and with a similar payload to below with status PREMOD  for both PENDING and UNMODERATED queues

event payload:

```
{
  id: 'd95d75ab-55d6-4d98-98a6-3f2980e7eaa5',
  createdAt: '2024-09-29T07:21:27.058Z',
  data: {
    queue: 'PENDING',
    commentID: 'c6ad8df2-b55d-4715-b4ef-68124e280f05',
    status: 'PREMOD',
    storyID: 'story-test',
    siteID: '373d38eb-b180-4772-b16f-85b14e87855b'
  },
  type: 'COMMENT_ENTERED_MODERATION_QUEUE',
  tenantID: '91d6e62e-ba7d-497d-8d0d-e4d56d48d33b',
  tenantDomain: 'localhost'
}
```

## Were any tests migrated to React Testing Library?

No

## How do we deploy this PR?

No special considerations.

### CODE REVIEW

File: server/WEBHOOKS.md [hunk 1]
```diff
@@ -230,6 +232,7 @@ function when comparing signatures.
 ```
 
 - <a id="comment-created-event">**COMMENT_CREATED**</a>
+  - only sent for published comments (ie status NONE or APPROVED)
```
Comment (ddrysdale-ffx) [inline server/WEBHOOKS.md:235]: decided to add these to the doco to clarify how these events work ... as we were caught out expecting these to be triggered for all comments

Comment (kabeaty) [reply server/WEBHOOKS.md:235]: Thank you! We will probably also want to update our `webhooks.md` in `docs`.

**Review by kabeaty — APPROVED** [review]
Looks good! Thank you for contributing this.

**Review by nick-funk — APPROVED** [review]

File: server/WEBHOOKS.md (hunk 1 above)
Comment (ddrysdale-ffx) [reply server/WEBHOOKS.md:235]: Ahh, I didn't realise there was another file. Let me apply the same change there.

Comment (ddrysdale-ffx) [reply server/WEBHOOKS.md:235]: Have now applied the same change to docs/docs/webhooks.md

**Comment by ddrysdale-ffx** [comment]
@kabeaty ... thanks for looking at this so quickly.

Just wanted to check a couple of things ...

1. There are some checks on the PR that don't appear to have completed successfully. Is there anything I need to do with that?
2. What would be the process for getting this into a Coral release?

**Pull request merged**